### PR TITLE
add pgd endpoints

### PR DESCRIPTION
**Pull request by massrb**
## Description of change

Add PGD endpoints

## Original issue(s)
department-of-veterans-affairs/va.gov-team#0000

### CODE REVIEW

File: modules/health_quest/app/controllers/health_quest/v0/pgd_ques_responses_controller.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+
+module HealthQuest
+  module V0
+    class PgdQuesResponsesController < HealthQuest::V0::BaseController
+      def show
```
Comment (erluti) [inline modules/health_quest/app/controllers/health_quest/v0/pgd_ques_responses_controller.rb:6]: This is probably just a stylistic difference, but I think we could drop some of these assignments and things are a little clearer like this:
```
      def show
        qresponse = ques_response_by_id
        render json: HealthQuest::V0::PGDQuesResponseSerializer.new(qresponse[:data], meta: qresponse[:meta])
      end
```

File: modules/health_quest/spec/request/pgd_questionnaires_request_spec.rb [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+      end
+    end
+  end
+
+  context 'health quest user' do
+    let(:current_user) { build(:user, :health_quest) }
+
+    describe 'GET questionnaires' do
+      it 'has access to questionnaires have access' do
+        get '/health_quest/v0/pgd_questionnaires/32'
```
Comment (erluti) [inline modules/health_quest/spec/request/pgd_questionnaires_request_spec.rb:35]: could there be some tests that describe the content of the responses?

File: modules/health_quest/app/serializers/health_quest/v0/pgd_questionnaire_serializer.rb [hunk 3]
```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+
+module HealthQuest
+  module V0
+    class PGDQuestionnaireSerializer
+      include FastJsonapi::ObjectSerializer
+
+      set_id do |object|
```
Comment (erluti) [inline modules/health_quest/app/serializers/health_quest/v0/pgd_questionnaire_serializer.rb:8]: the id in FastJonapi use the `:id` key by default, so the `set_id` should be unnecessary here

File: modules/health_quest/config/routes.rb [hunk 4]
```diff
@@ -4,5 +4,9 @@
   namespace :v0, defaults: { format: :json } do
     resources :appointments, only: %i[index show] do
     end
+    resources :pgd_questionnaires, only: %i[show] do
```
Comment (erluti) [inline modules/health_quest/config/routes.rb:7]: no block is necessary here, this could be just `resources :pgd_questionnaires, only: %i[show]`

**Review by erluti — COMMENTED** [review]
I have some refactoring suggestions.  Most of them apply to the both controllers

File: modules/health_quest/config/routes.rb (hunk 4 above)
Comment (massrb) [reply modules/health_quest/config/routes.rb:7]: ok, I have addressed the requested changes

File: modules/health_quest/app/controllers/health_quest/v0/pgd_questionnaires_controller.rb [hunk 5]
```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+
+module HealthQuest
+  module V0
+    class PgdQuestionnairesController < HealthQuest::V0::BaseController
+      def show
+        ques = questionnaire_by_id
```
Comment (erluti) [inline modules/health_quest/app/controllers/health_quest/v0/pgd_questionnaires_controller.rb:7]: I think there could be a similar refactoring here https://github.com/department-of-veterans-affairs/vets-api/pull/5001/files#r497520063

File: modules/health_quest/config/pgd_ques_responses_controller.rb [hunk 6]
```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+
+module HealthQuest
+  module V0
+    class PgdQuesResponsesController < HealthQuest::V0::BaseController
+      def show
+        qresponse = ques_response_by_id
```
Comment (erluti) [inline modules/health_quest/config/pgd_ques_responses_controller.rb:7]: and here

File: modules/health_quest/spec/request/pgd_questionnaires_request_spec.rb (hunk 2 above)
Comment (erluti) [reply modules/health_quest/spec/request/pgd_questionnaires_request_spec.rb:35]: thanks!

**Comment by LindseySaari** [comment]
Do you have swagger doc updates for this PR?

File: modules/health_quest/app/controllers/health_quest/v0/pgd_questionnaires_controller.rb (hunk 5 above)
Comment (dillo) [reply modules/health_quest/app/controllers/health_quest/v0/pgd_questionnaires_controller.rb:7]: Refactored as requested.

File: modules/health_quest/config/pgd_ques_responses_controller.rb (hunk 6 above)
Comment (dillo) [reply modules/health_quest/config/pgd_ques_responses_controller.rb:7]: Refactored as requested.

**Comment by erluti** [comment]
the pipeline failure should be fixed if you pull in the latest master

**Comment by dillo** [comment]
> Do you have swagger doc updates for this PR?

Hi @LindseySaari I'm the new backend engineer for the Health Care team and I just took over the code-base for it and this story/ticket/branch. I believe that the swagger docs are being reviewed in this branch: https://github.com/department-of-veterans-affairs/vets-api/pull/4937 I will be taking over ownership of this as well.

**Comment by dillo** [comment]
> the pipeline failure should be fixed if you pull in the latest master

@erluti Done! I do wish to clean up the code a little further. Can I do that here or should I just start another PR based on this branch? Thanks!

**Comment by erluti** [comment]
> I do wish to clean up the code a little further. Can I do that here or should I just start another PR based on this branch? Thanks!

If it's scoped to the work this change is trying to accomplish, go ahead and make this code its best.  But if it's other refactoring for nearby code, that should be its own branch.

**Review by dillo — APPROVED** [review]

**Comment by dillo** [comment]
Not sure how but I was looking at a different PR and this got approved instead by mistake. Sorry about that.

**Comment by erluti** [comment]
> Not sure how but I was looking at a different PR and this got approved instead by mistake. Sorry about that.

no worries.  does this mean it's ready for final review?

**Comment by dillo** [comment]
> > Not sure how but I was looking at a different PR and this got approved instead by mistake. Sorry about that.
> 
> no worries. does this mean it's ready for final review?

@erluti I don't think this is quite ready yet. For example line, https://github.com/department-of-veterans-affairs/vets-api/pull/5001/files#diff-a0e7c3c36dc46c3272f33182e0f914e8R23 would throw an exception if the `result` object is `nil` or an `array` or is simply not a `hash`.

It looks like the ticket that this story is based on is a bit ambiguous on the expectations. As such I want to narrow the focus and scope of this PR and just have the endpoints return a status of 200 for now.

I will comment again on this when it's ready for *final* review. Thank you for your time and patience.

**Comment by dillo** [comment]
Can we close this PR? I'm breaking this out into two separate PR's to address the two basic endpoints. Here's the first one: https://github.com/department-of-veterans-affairs/vets-api/pull/5047 cc: @erluti

**Comment by dillo** [comment]
Closing this PR as it's no longer needed!

**Pull request closed without merging**